### ISSUE TITLE
The current implementation can miss interface on windows platform

### ISSUE THREAD

**Issue by xmh0511** [mmastrac/getifaddrs#3]
https://github.com/mmastrac/getifaddrs/blob/master/src/lib.rs#L425
````rust
if self.current_unicast.is_null() {
    continue;
}
````
This test can be true for some network interfaces whose IP is set by command such that this network interface is omitted in the result. Specifically, tun/tap interface created by wintun and tap-windows.

````rust
        let mut size = 0;
        let flags = GAA_FLAG_INCLUDE_PREFIX | GAA_FLAG_INCLUDE_GATEWAYS;
        let family = AF_UNSPEC as u32;

        // Make an initial call to GetAdaptersAddresses to get the
        // size needed into the size variable
        let result = unsafe {
            GetAdaptersAddresses(
                family,
                flags,
                std::ptr::null_mut(),
                std::ptr::null_mut(),
                &mut size,
            )
        };

        if result != ERROR_BUFFER_OVERFLOW {
            return Err(format!(
                "GetAdaptersAddresses failed: {}",
                format_message(result).map_err(|v| v.to_string())?
            )
            .into());
        }
        // Allocate memory for the buffer
        let mut addresses: Vec<u8> = vec![0; (size + 4) as usize];

        // Make a second call to GetAdaptersAddresses to get the actual data we want
        let result = unsafe {
            let addr = addresses.as_mut_ptr() as *mut IP_ADAPTER_ADDRESSES_LH;
            GetAdaptersAddresses(family, flags, std::ptr::null_mut(), addr, &mut size)
        };

        if ERROR_SUCCESS != result {
            return Err(format!(
                "GetAdaptersAddresses failed: {}",
                format_message(result).map_err(|v| v.to_string())?
            )
            .into());
        }

        // If successful, output some information from the data we received
        let mut all_address = vec![];
        let mut current_addresses = addresses.as_ptr() as *const IP_ADAPTER_ADDRESSES_LH;
        while !current_addresses.is_null() {
            unsafe {
                let adapter = &*current_addresses;
                let mut current_address = adapter.FirstUnicastAddress;
                while !current_address.is_null() {
                    match convert_to_interface(adapter, &*current_address) {
                        Ok(interface) => {
                            all_address.push(interface);
                        }
                        Err(e) => {
                            log::error!("Failed to parse address: {}", e);
                        }
                    }
                    current_address = (*current_address).Next;
                }
                current_addresses = (*current_addresses).Next;
            }
        }
````
while this approach can acquire these network interfaces that will be omitted by this crate and whose addresses are correctly acquired.

**Comment by mmastrac** [comment]
Fixed by PR #4

**Comment by mmastrac** [comment]
Released as 0.1.4
